README.md

```diff
@@ -0,0 +1,145 @@
+
+- 🟨 `GET: /api/books` - **Retrieve Book Catalog**: Users can browse the catalog of available books.
+- 🟨 `GET: /api/books/{id}` - **View Book Details**: Users can also check information about each book individually.
+- 🟨 `GET: /api/books/search` - **Search Books**: Users can search for books based on various parameters.
+- 🟥 `POST: /api/books` - **Add New Book**: Admin users can add new books to the catalog.
+- 🟥 `PUT: /api/books/{id}` - **Update Book Details**: Admin users can update existing book details.
+- 🟥 `DELETE: /api/books/{id}` - **Delete Book**: Admin users can remove books from the catalog.
+
+### Category Management
+
+- 🟨 `GET: /api/categories` - **Retrieve Categories**: Users can view the list of available categories.
+- 🟨 `GET: /api/categories/{id}` - **View Category**: Users can also check information about each category individually.
+- 🟨 `GET: /api/categories/{id}/books` - **Get Books by Category**: Users can retrieve all books belonging to a specific category.
+- 🟥 `POST: /api/categories` - **Create Category**: Admin users can create new categories for books.
+- 🟥 `PUT: /api/categories/{id}` - **Update Category**: Admin users can update category details.
+- 🟥 `DELETE: /api/categories/{id}` - **Delete Category**: Admin users can delete categories.
+
+### Shopping Cart
+
+- 🟨 `GET: /api/cart` - **View Cart**: Users can view the contents of their shopping cart.
+- 🟨 `POST: /api/cart` - **Add to Cart**: Users can add books to their shopping cart.
+- 🟨 `PUT: /api/cart/cart-items/{id}` - **Update Cart**: Users can update the quantity of a specific item in their cart.
+- 🟨 `DELETE: /api/cart/cart-items/{id}` - **Remove from Cart**: Users can remove a specific item from their cart.
+
+### Order Management
+
+- 🟨 `GET: /api/orders` - **View Orders History**: Users can view all their orders history.
+- 🟨 `GET: /api/orders/{id}/items` - **View Order Items**: Users can view all items included in a specific order.
+- 🟨 `GET: /api/orders/{orderId}/items/{itemId}` - **View Specific Order Item**: Users can check details of a particular item within an order.
+- 🟨 `POST: /api/orders` - **Place Order**: Users can place orders for the books in their cart.
+- 🟥 `PATCH: /api/orders/{id}` - **Update Order Status**: Admin users can update the status of orders.
+
+## SQL Database Diagram
+Below is a representation of the database used in the project:
+
+![bookstore-db-diagram](bookstore-db-diagram.png)
+
+## How to run Book Store API
+1. Install [Docker](https://www.docker.com/products/docker-desktop/)
+2. Clone the project [git repository](https://github.com/QbaSekowski/spring-book-app.git)
+3. In the .env file you should provide necessary DB and Docker variables, here is an example:
+```mysql
+MYSQLDB_USER=root  
+MYSQLDB_ROOT_PASSWORD=root  
+MYSQLDB_DATABASE=test_db  
+MYSQLDB_LOCAL_PORT=3307  
+MYSQLDB_DOCKER_PORT=3306  
+SPRING_LOCAL_PORT=8088  
+SPRING_DOCKER_PORT=8080  
+DEBUG_PORT=5005
+```
+4. Run the command `mvn clean package`
```

Comment:
Unless you're building a package inside the container, you should also list JDK and Maven in step 1